src/parquet_writer.h

```diff
@@ -3,90 +3,130 @@
+  case arrow::Type::type::INT16:
+    return MakeField(name, arrow::int16());
 
-  case FieldType::Type::DOUBLE:
-    return parquet::ConvertedType::NONE;
+  case arrow::Type::type::UINT32:
+    return MakeField(name, arrow::uint32());
 
-  case FieldType::Type::BOOLEAN:
-    return parquet::ConvertedType::NONE;
+  case arrow::Type::type::INT32:
+    return MakeField(name, arrow::int32());
 
-  case FieldType::Type::UTF8:
-    return parquet::ConvertedType::UTF8;
-  }
+  case arrow::Type::type::UINT64:
+    return MakeField(name, arrow::uint64());
 
-  // Execution only reaches here if you do something unsafe, but the compiler is complaining
-  return parquet::ConvertedType::UNDEFINED;
-}
+  case arrow::Type::type::INT64:
+    return MakeField(name, arrow::int64());
 
-static parquet::Type::type LogicalTypeFromFieldType(FieldType::Type type) {
-  switch (type) {
-  case FieldType::Type::INT32:
-    return parquet::Type::INT32;
+  case arrow::Type::type::FLOAT:
+    return MakeField(name, arrow::float32());
 
-  case FieldType::Type::INT64:
-    return parquet::Type::INT64;
+  case arrow::Type::type::DOUBLE:
+    return MakeField(name, arrow::float64());
 
-  case FieldType::Type::TIMESTAMP_MICROS:
-    return parquet::Type::INT64;
+  case arrow::Type::type::STRING:
+    return MakeField(name, arrow::utf8());
 
-  case FieldType::Type::TIMESTAMP_MILLIS:
-    return parquet::Type::INT64;
+  case arrow::Type::type::BINARY:
+    return MakeField(name, arrow::binary());
 
-  case FieldType::Type::DATE32:
-    return parquet::Type::INT32;
+  case arrow::Type::type::FIXED_SIZE_BINARY: {
+    auto width = obj.Get("width").ToNumber().Int32Value();
+    return MakeField(name, arrow::fixed_size_binary(width));
+  }
 
-  case FieldType::Type::DOUBLE:
-    return parquet::Type::DOUBLE;
+  case arrow::Type::type::DATE32:
+    return MakeField(name, arrow::date32());
 
-  case FieldType::Type::BOOLEAN:
-    return parquet::Type::BOOLEAN;
+  case arrow::Type::type::TIMESTAMP: {
+    auto unit = static_cast<arrow::TimeUnit::type>(obj.Get("unit").ToNumber().Int32Value());
+    return MakeField(name, arrow::timestamp(unit));
+  }
+
+  case arrow::Type::type::TIME32: {
+    auto unit = static_cast<arrow::TimeUnit::type>(obj.Get("unit").ToNumber().Int32Value());
+    return MakeField(name, arrow::time32(unit));
+  }
 
-  case FieldType::Type::UTF8:
-    return parquet::Type::BYTE_ARRAY;
+  case arrow::Type::type::TIME64: {
+    auto unit = static_cast<arrow::TimeUnit::type>(obj.Get("unit").ToNumber().Int32Value());
+    return MakeField(name, arrow::time64(unit));
   }
 
-  // Execution only reaches here if you do something unsafe, but the compiler is complaining
-  return parquet::Type::UNDEFINED;
+  default:
+    // Should only happen if JS users use a number instead of the enum
+    throw std::runtime_error("Invalid type id");
+  }
+}
+
+inline static ArrowFieldPtr NapiObjToArrowField(const std::string& name, const Napi::Object& obj) {
```

Comment:
This is quite convoluted. Overloads should be avoided. Also why not just have the version with 2 parameters? You don't need the 3 parameters version.

Reply:
Also avoid `obj`. Spell it out loud, like it is in the rest of the APIs used in this PR.